Sources/Redis/Application+Redis.swift

```diff
@@ -0,0 +1,74 @@
+import Vapor
+
+extension Application {
+    public var redis: Redis {
+        .init(application: self)
+    }
+
+    public struct Redis {
+        struct ConfigurationKey: StorageKey {
+            typealias Value = RedisConfiguration
+        }
+
+        public var configuration: RedisConfiguration? {
+            get {
+                self.application.storage[ConfigurationKey.self]
+            }
+            nonmutating set {
+                if self.application.storage.contains(PoolKey.self) {
+                    fatalError("Cannot modify your Redis configuration after redis has been used")
```

Comment:
I think you definitely want to fatalError here and not warning, at least in Redis case, for security reasons.

Imagine you have a client that uses a default redis for most things, but has a special redis for secure and or confidential data.

The user might think he can simply swap the configuration during runtime at random times to write to secure redis, but if we keep this scenario as a warning, a user may end up writing confidential data to the wrong redis. 

By the time the warning is taken into account, the damage may already have been done. Simply ignoring is not an option here.